src/features/defi/providers/idle/components/IdleManager/Overview/IdleEmpty.tsx

```diff
@@ -0,0 +1,218 @@
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { selectSupportsFiatRampByAssetId } from 'state/apis/fiatRamps/selectors'
+import { IdleTag } from 'state/slices/opportunitiesSlice/resolvers/idle/constants'
+import type { TagDescription } from 'state/slices/opportunitiesSlice/types'
+import { selectAssetById, selectPortfolioCryptoHumanBalanceByFilter } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+const idleTagDescriptions: Record<IdleTag, TagDescription[]> = {
+  [IdleTag.BestYield]: [
+    {
+      title: 'idle.emptyBody.bestYield.page-1.title',
+      description: 'idle.emptyBody.bestYield.page-1.body',
+      icon: <Image src={BestYieldIcon} />,
+    },
+    {
+      title: 'idle.emptyBody.bestYield.page-2.title',
+      bullets: ['idle.emptyBody.monitorSystem', 'idle.emptyBody.offerHighReturn'],
+    },
+  ],
+  [IdleTag.JuniorTranche]: [
+    {
+      title: 'idle.emptyBody.juniorTranche.page-1.title',
+      description: 'idle.emptyBody.juniorTranche.page-1.body',
+      icon: <Image src={JuniorTrancheIcon} />,
+    },
+    {
+      title: 'idle.emptyBody.juniorTranche.page-2.title',
+      bullets: [
+        'idle.emptyBody.leverageYield',
+        'idle.emptyBody.higherRisk',
+        'idle.emptyBody.noLockingPeriod',
+        'idle.emptyBody.autoReinvest',
+      ],
+    },
+  ],
+  [IdleTag.SeniorTranche]: [
+    {
+      title: 'idle.emptyBody.seniorTranche.page-1.title',
+      description: 'idle.emptyBody.seniorTranche.page-1.body',
+      icon: <Image src={SeniorTranceIcon} />,
+    },
+    {
+      title: 'idle.emptyBody.seniorTranche.page-2.title',
+      bullets: [
+        'idle.emptyBody.minRisk',
+        'idle.emptyBody.builtInCoverage',
+        'idle.emptyBody.noLockingPeriod',
+        'idle.emptyBody.autoReinvest',
+      ],
+    },
+  ],
+}
+
+type IdleEmptyProps = {
+  assetId: AssetId
+  onClick?: () => void
+  tags?: string[]
+  apy: string
+}
+
+export const IdleEmpty = ({ assetId, onClick, tags, apy }: IdleEmptyProps) => {
+  const translate = useTranslate()
+  const { open: openFiatRamp } = useModal().fiatRamps
+  const asset = useAppSelector(state => selectAssetById(state, assetId))
+  const filter = useMemo(() => ({ assetId }), [assetId])
+  const bgImage = useColorModeValue('none', IdleBg)
+  const assetSupportsBuy = useAppSelector(s => selectSupportsFiatRampByAssetId(s, filter))
+  const cryptoBalance =
+    useAppSelector(state => selectPortfolioCryptoHumanBalanceByFilter(state, filter)) ?? '0'
+  const textShadow = useColorModeValue(
+    '--chakra-colors-blackAlpha-50',
+    '--chakra-colors-blackAlpha-400',
+  )
+
+  const handleAssetBuyClick = useCallback(() => {
+    openFiatRamp({ assetId, fiatRampAction: FiatRampAction.Buy })
+  }, [assetId, openFiatRamp])
+
+  const renderFooter = useMemo(() => {
+    return (
+      <Flex flexDir='column' gap={4} width='full'>
+        {bnOrZero(cryptoBalance).eq(0) && assetSupportsBuy && (
```

Comment:
I'd abstract the very left side of this (`bnOrZero(cryptoBalance).eq(0)`) to its own variable, function calls in render methods are usually an antipattern for memoization purposes - granted this is already a memoized node so it is actually fine, but always better to follow best practices